privacyidea/api/lib/postpolicy.py

```diff
@@ -721,10 +739,13 @@ def autoassign(request, response):
                                      "info":
                                          "Token assigned via auto assignment",
                                      "serial": token_obj.token.serial})
+                                # The token was assigned by autoassign. We save the policy name
+                                g.audit_object.add_policy(autoassign_values.values()[0])
```

Comment:
Please use `list(autoassign_values.values())[0]` for Python 3 compatibility.

Reply:
``.values()`` also returns an iterator in python3? 
This is why we need to create a list?

Reply:
Why would we do this? Is it faster or saves it _memory_ (I would guess this)?

Reply:
It is faster and i think it saves memory as well (didn't test):
Python 2:
> In [318]: %timeit next(iter(D.values()))
The slowest run took 29.10 times longer than the fastest. This could mean that an intermediate result is being cached.
1000000 loops, best of 3: 270 ns per loop

> In [319]: %timeit list(D.values())[0]
The slowest run took 7.90 times longer than the fastest. This could mean that an intermediate result is being cached.
1000000 loops, best of 3: 1.15 µs per loop

Python 3:
> In [240]: %timeit next(iter(D.values()))                                                                                                                                                      
180 ns ± 0.437 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

> In [241]: %timeit list(D.values())[0]                                                                                                                                                         
889 ns ± 8.05 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

Reply:
Thanks. Done in commit 25e4552

Reply:
BTW the same is true for `.keys()`... But no need to fix it now, we can improve performance later ;-)